### PR TITLE
Edit-Form component added

### PR DESCRIPTION
**Pull request by shubhwale**

### CODE REVIEW

File: src/app/shared/book.service.ts [hunk 1]
```diff
@@ -20,8 +20,37 @@ import { catchError } from 'rxjs/operators';
...
-      return this.http.get(this.BaseUri+'/books/'+id).toPromise();
+    findBookById(id: number) : Promise<Object> {
+      return this.http.get(this.BaseUri+'/books/'+id).pipe(catchError(this.handleError)).toPromise();
+    }
+
+    addBook(bookData: object) : Promise<Object> {
+      return this.http.post(this.BaseUri+'/books/addbook',bookData,{observe : 'response'}).pipe(catchError(this.handleError)).toPromise();
+    }
+
+    editBook(bookData : any) : Promise<Object> {
```
Comment (bala0105) [inline src/app/shared/book.service.ts:31]: type any

File: src/app/shared/book.service.ts [hunk 2]
```diff
@@ -20,8 +20,37 @@ import { catchError } from 'rxjs/operators';
...
+      return this.http.put(this.BaseUri+'/books/'+bookData.bookId,bookData,{observe : 'response'}).pipe(catchError(this.handleError)).toPromise();
+    }
+
+    generateImageURL(title : any) : string {
```
Comment (bala0105) [inline src/app/shared/book.service.ts:35]: type any

**Comment by ketannaik** [comment]
@shubhwale : You should have merged the previous branch first i.e. AddBooks then raise the PR for this branch. This branch has lot of similar code from Add Books branch. 
In actual projects, this will be an overhead for the reviewers. So I would suggest complete, the Add Books and then merge it in this branch

File: src/app/shared/book.service.ts (hunk 1 above)
Comment (shubhwale) [reply src/app/shared/book.service.ts]: done

File: src/app/shared/book.service.ts (hunk 2 above)
Comment (shubhwale) [reply src/app/shared/book.service.ts]: done

**Pull request merged**